### PR TITLE
prevent anyone from accidentally setting the blocksize to zero

### PR DESCRIPTION
**Pull request by deitch**
Although we do have a default, in theory, but if someone actually calls: `WithBlocksize(0)`, it will override the default, and break all sorts of things. This adds a safety check.

### CODE REVIEW

File: pkg/content/opts.go [hunk 1]
```diff
@@ -47,10 +47,13 @@ func WithOutputHash(hash digest.Digest) WriterOpt {
...
 
 // WithBlocksize set the blocksize used by the processor of data.
-// The default is DefaultBlocksize, which is the same as that used by io.Copy
+// The default is DefaultBlocksize, which is the same as that used by io.Copy.
+// Includes a safety check to ensure the caller doesn't actively set it to <= 0.
 func WithBlocksize(blocksize int) WriterOpt {
 	return func(w *WriterOpts) error {
-		w.Blocksize = blocksize
+		if blocksize > 0 {
+			w.Blocksize = blocksize
```
Comment (jdolitsky) [inline pkg/content/opts.go:55]: Should we check for inverse and just throw an error here?

**Review by jdolitsky — CHANGES_REQUESTED** [review]

File: pkg/content/opts.go (hunk 1 above)
Comment (deitch) [reply pkg/content/opts.go:55]: I don't understand.

Comment (deitch) [reply pkg/content/opts.go:55]: Do you mean, rather than ignoring/fixing it if the `blocksize <= 0`, treat `blocksize <= 0` as an error?

Comment (jdolitsky) [reply pkg/content/opts.go:55]: > treat blocksize <= 0 as an error?

Yes

Comment (deitch) [reply pkg/content/opts.go]: Done.

**Review by jdolitsky — APPROVED** [review]

**Pull request merged**